src/main/kotlin/FormulaParser.kt

```diff
@@ -0,0 +1,49 @@
+class FormulaParser {
+    companion object {
+        const val DIVISION_CHARACTER = ' '
+    }
+
+    fun parse(formula: String): List<String> {
+        val formulaComponents = splitFormula(formula)
+
+        validateFormula(formulaComponents)
+
+        return formulaComponents
+    }
+
+    private fun splitFormula(formula: String) = formula.split(DIVISION_CHARACTER)
+
+    private fun validateFormula(formulaComponents: List<String>) {
+        validateNotBlank(formulaComponents)
+        validateFormulaStructure(formulaComponents)
+        validateNumbers(formulaComponents)
+        validateOperators(formulaComponents)
+    }
+
+    private fun validateNotBlank(formulaComponents: List<String>) {
+        if (formulaComponents.filterNot { it.isBlank() }.isEmpty()) {
```

Comment:
p3; 모든 component가 blank일 때 조건에 통과하는 거라면,
```suggestion
        if (formulaComponents.all { it.isBlank() }) {
```
kotlin의 all을 활용해 볼 수 있을 거 같아요. 😊